### PR TITLE
Use composition in ObjectHypothesisWithPose

### PR DESCRIPTION
**Pull request by mintar**

### CODE REVIEW

File: msg/ObjectHypothesisWithPose.msg [hunk 1]
```diff
@@ -1,18 +1,12 @@
...
-# The unique ID of the object class. To get additional information about
-#   this ID, such as its human-readable class name, listeners should perform a
-#   lookup in a metadata database. See vision_msgs/VisionInfo.msg for more detail.
-string id
-
-# The probability or confidence value of the detected object. By convention,
-#   this value should lie in the range [0-1].
-float64 score
+# The object hypothesis (ID and score).
+ObjectHypothesis hypothesis
```
Comment (SteveMacenski) [inline msg/ObjectHypothesisWithPose.msg:4]: I'd like a different name for this, but honestly I can't think of a better option

**Review by SteveMacenski — APPROVED** [review]
@Kukanani please review as well

**Review by Kukanani — APPROVED** [review]
This looks good to me. @SteveMacenski should we look at making a new branch for ros2 development, to distinguish it from Foxy and earlier?

**Comment by SteveMacenski** [comment]
Oh, very good point. `foxy` branch now made, its safe to merge this

**Pull request merged**